Tests/ParseSwiftTests/ParseObjectBatchTests.swift

```diff
@@ -0,0 +1,722 @@
+            } catch {
+                return nil
+            }
+        }
+
+        do {
+            let saved = try [score, score2].saveAll()
+
+            XCTAssertEqual(saved.count, 2)
+            XCTAssertThrowsError(try saved[0].get())
+            XCTAssertThrowsError(try saved[1].get())
+
+        } catch {
+            XCTFail(error.localizedDescription)
+        }
+
+        do {
+            let saved = try [score, score2].saveAll(options: [.useMasterKey])
+
+            XCTAssertEqual(saved.count, 2)
+            XCTAssertThrowsError(try saved[0].get())
+            XCTAssertThrowsError(try saved[1].get())
+
+        } catch {
+            XCTFail(error.localizedDescription)
+        }
+    }
+
+    func testUpdateAll() { // swiftlint:disable:this function_body_length cyclomatic_complexity
+        var score = GameScore(score: 10)
+        score.objectId = "yarr"
+        score.createdAt = Calendar.current.date(byAdding: .init(day: -1), to: Date())
+        score.updatedAt = Calendar.current.date(byAdding: .init(day: -1), to: Date())
+        score.ACL = nil
+
+        var score2 = GameScore(score: 20)
+        score2.objectId = "yolo"
+        score2.createdAt = Calendar.current.date(byAdding: .init(day: -1), to: Date())
+        score2.updatedAt = Calendar.current.date(byAdding: .init(day: -1), to: Date())
+        score2.ACL = nil
+
+        var scoreOnServer = score
+        scoreOnServer.updatedAt = Date()
+        var scoreOnServer2 = score2
+        scoreOnServer2.updatedAt = Date()
+
+        let response = [BatchResponseItem<GameScore>(success: scoreOnServer, error: nil),
+        BatchResponseItem<GameScore>(success: scoreOnServer2, error: nil)]
+
+        MockURLProtocol.mockRequests { _ in
+            do {
+                let encoded = try scoreOnServer.getEncoderWithoutSkippingKeys().encode(response)
+                return MockURLResponse(data: encoded, statusCode: 200, delay: 0.0)
+            } catch {
+                return nil
+            }
+        }
+        do {
+            let saved = try [score, score2].saveAll()
+
+            XCTAssertEqual(saved.count, 2)
+            switch saved[0] {
+
+            case .success(let first):
+
+                guard let savedCreatedAt = first.createdAt,
+                    let savedUpdatedAt = first.updatedAt else {
+                        XCTFail("Should unwrap dates")
+                        return
+                }
+                guard let originalCreatedAt = score.createdAt,
+                    let originalUpdatedAt = score.updatedAt else {
+                        XCTFail("Should unwrap dates")
+                        return
+                }
+                /*Date's are not exactly as their original because the URLMocking doesn't use the same dateEncoding
+                 strategy, so we only compare the day*/
+                XCTAssertTrue(Calendar.current.isDate(savedCreatedAt, equalTo: originalCreatedAt, toGranularity: .day))
```

Comment:
Can you elaborate on this a little? Why / how are they not the same, and if we can't work around that at all, can we use a finer granularity? A day seems like a really wide range.

Reply:
Since I’m mocking in the test, I’m attempting to create responses the server would have, but don’t have full access to how it creates things like dates. In some cases, I can just use the encoder/decoder in the framework. If you attempt to completely compare the dates directly here, the tests will fail, I believe because of the reasons I mentioned.

I don’t believe this test needs to be more granular as I specify the difference in days for the test case so it’s the only thing that needs to be compared

Reply:
Your comment made me look back at this and I fixed the problem. The issue was `fetch` never used `apply` after it decodes so the dates wasn't being applied correctly